include/libdnf/utils/weak_ptr.hpp

```diff
@@ -216,6 +216,7 @@ struct WeakPtr {
     /// Checks if the other WeakPtr instance has the same WeakPtrGuard.
     bool has_same_guard(const WeakPtr & other) const noexcept { return guard == other.guard; }
 
+    TPtr * operator*() const { return get(); }
```

Comment:
Operator `*` is used for dereferencing pointer. For getting pointer is `get()`.
Try:
`TPtr & operator*() const { return *get(); }`

Then you replace `CPPUNIT_ASSERT(**first > **second);` by `CPPUNIT_ASSERT(*first > *second);` in unittests

Reply:
fixed